### DISCUSSION TITLE
Deadlock after "run"-function in the tutorial

### DISCUSSION THREAD

**Discussion by JoDrexler** [lava-nc/lava#516] [Q&A]
I am new to Lava and started with the end_to_end tutorial "tutorial00_tour_through_lava". It contains two sections. Calling the "run"-function in the first section works perfectly fine. Calling it in the second section keeps the program on running infinitely without throwing an error. Also experienced this problem in other tutorials of the "in_depth" directory. I was first thinking that there are some issues with my installation, but the "run"-function works for some tutorials. Does anybody have an idea how to solve my problem?
Thanks for your help!

**Comment by weidel-p** [comment]
Hi, thanks for your message. I'm sorry the tutorials don't work for you out of the box.

There are some known issues with running a network in a notebook under Windows (i.e. https://github.com/lava-nc/lava/issues/401). May I ask you which operating system you use?

If that's the issue, you can still use Lava without notebooks. Could you please try to export and run the notebook as .py scripts?

**Reply by JoDrexler** [reply]
Hi, first of all, thanks a lot for your answer and sorry for the delayed reply.
I am using Windows.

I tried to run tutorials as .py scripts, but it was not successful:
1. I got a RuntimeError for the first part of the "tutorial00_tour_through_lava" which already worked in the notebook.
2. For tutorial02 and tutorial03, I got a ValueError (_\_\_mp_main\_\_.\_\_spec\_\_ is None_).

Do you have any idea what can cause these issues?

**Reply by weidel-p** [reply]
Could you paste the complete error traces here? And maybe add the scripts to exported from the tutorials?

**Reply by JoDrexler** [reply]
Thanks a lot for your help! I coppied the error traces into txt-files. Let me know if you need further information.

[ERROR-deadlock.zip](https://github.com/lava-nc/lava/files/10214884/ERROR-deadlock.zip)
